lib/oban/queue/producer.ex

```diff
@@ -100,7 +100,7 @@ defmodule Oban.Queue.Producer do
     # block the producer while awaiting a retry.
     Task.Supervisor.async_nolink(state.foreman, fn ->
       Breaker.with_retry(fn ->
-        %{exec | kind: :error, error: error, stacktrace: stack, state: :failure}
+        %{exec | kind: {:EXIT, pid}, error: error, stacktrace: stack, state: :failure}
```

Comment:
The `kind` is exposed through telemetry data, and meant to match the values that come from `telemetry.span`. What if we keep this as `:error` but ensure that `stack` is an empty list?

Reply:
Options:
1. kind: `:error`: `"error" => "** (ErlangError) Erlang error: :timeout"`
2. kind: `:exit` (also a valid `kind` according to `telemetry` docs): `"error" => "** (exit) time out"`
3. kind: `{:EXIT, pid}`: `"error" => "** (EXIT from #PID<0.3249.0>) time out" `

I'm not quite sure what happens in Telemetry when you pass `{:EXIT, pid}`.

Reply:
I looked into this further, `Exception.kind` has type `:error | :exit | :throw | {:EXIT, pid}`, whilst telemetry kind only includes `:error | :exit | :throw`. I think it's a bit weird. We are using `kind` both in calls to the `Exception` module and in calls to telemetry.

I added a little case statement to convert from `Exception.kind` to the `metadata.kind` that `:telemetry` is expecting. This gives us the best error message, and still conforms to telemetry's spec.

Let me know if you'd still prefer `:exit`.